File: app/src/at-myhuynh/java/asiantech/internship/summer/java_core/ex20/FibonacciPrime.java
package asiantech.internship.summer.java_core.ex20;

import java.util.logging.Logger;

import asiantech.internship.summer.java_core.common.Common;

public class FibonacciPrime {
    public static void main(String[] args) {
        int n = Common.input("Nhập vào số n:");
        showFibonacciIsPrime(n);
    }

    private static void showFibonacciIsPrime(int n) {
        Logger log = Logger.getLogger(FibonacciPrime.class.getName());
        int f1 = 1;
        int f2 = 1;
        int fn = 1;

        do {
            if (fn < n && Common.isPrime(fn)) {
                log.info(String.valueOf(fn));
            }
            fn = f1 + f2;
            f1 = f2;
            f2 = fn;
        } while (fn < n);
    }
}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/ex20/FibonacciPrime.kt
package asiantech.internship.summer.kotlin.ex20

import asiantech.internship.summer.kotlin.common.Common
import asiantech.internship.summer.kotlin.common.isPrime

object FibonacciPrime {
    @JvmStatic
    fun main(args: Array<String>) {
        val n = Common.input("Nhập vào số n:")
        getListFibonacciLessThenN(n).filter(Int::isPrime).forEach(::println)
    }

    private fun getListFibonacciLessThenN(n: Int): MutableList<Int> {
        val listFibonacci = mutableListOf<Int>()
        var t1 = 0
        var t2 = 1
        while (t1 <= n) {
            listFibonacci.add(t1)
            val sum = t1 + t2
            t1 = t2
            t2 = sum
        }

        return listFibonacci
    }
}
